packages/klighd-interactive/src/layered/relativeConstraint-utils.ts

```diff
@@ -0,0 +1,211 @@
+            }
+            break;
+        }
+        case Direction.UP:
+        case Direction.DOWN: {
+            if (succ === undefined && pred !== undefined && pred.id !== targetNode.id) {
+                iLSuccOf = true
+            } else if (succ !== undefined && pred !== undefined) {
+                iLSuccOf = Math.abs(midX - (predX + pred.size.width)) < Math.abs(midX - succX)
+                iLPredOf = Math.abs(midX - (predX + pred.size.width)) > Math.abs(midX - succX)
+            } else if (pred === undefined && succ !== undefined && succ.id !== targetNode.id) {
+                iLPredOf = true
+            }
+            break;
+        }
+    }
+
+    // Check whether the target is allowed to be successors/predecessor of the preceding/succeeding node.
+    if (iLSuccOf) {
+        if (!forbiddenRC(targetNode, pred)) {
+            return new RelConsData(RelCons.IN_LAYER_SUCC_OF, pred, targetNode)
+        }
+    } else if (iLPredOf) {
+        if (!forbiddenRC(targetNode, succ)) {
+            return new RelConsData(RelCons.IN_LAYER_PRED_OF, succ, targetNode)
+        }
+    }
+    // If no successor or predecessor exist 
+    return new RelConsData(RelCons.UNDEFINED, targetNode, targetNode)
+}
+
+/**
+ * Determines the nodes that are connected to {@code node} by relative constraints.
+ * The nodes are not sorted.
+ * @param node One node of the chain
+ * @param layerNodes Nodes that are in the same layer as {@code node}
+ */
+export function getChain(node: KNode, layerNodes: KNode[]): KNode[] {
+    layerNodes.sort((a, b) => a.properties['org.eclipse.elk.layered.crossingMinimization.positionId'] as number
+        - (b.properties['org.eclipse.elk.layered.crossingMinimization.positionId'] as number))
+    const pos = layerNodes.indexOf(node)
+    const chainNodes: KNode[] = []
+    chainNodes[0] = node
+    // from node to the start
+    for (let i = pos - 1; i >= 0; i--) {
+        if (layerNodes[i].properties['org.eclipse.elk.layered.crossingMinimization.inLayerPredOf'] != null
+            || layerNodes[i + 1].properties['org.eclipse.elk.layered.crossingMinimization.inLayerSuccOf'] != null) {
+            chainNodes[chainNodes.length] = layerNodes[i]
+        } else {
+            i = -1
+        }
+    }
+    // from node to the end
+    for (let i = pos + 1; i < layerNodes.length; i++) {
+        if (layerNodes[i].properties['org.eclipse.elk.layered.crossingMinimization.inLayerSuccOf'] != null
+            || layerNodes[i - 1].properties['org.eclipse.elk.layered.crossingMinimization.inLayerPredOf'] != null) {
+            chainNodes[chainNodes.length] = layerNodes[i]
+        } else {
+            i = layerNodes.length
+        }
+    }
+
+    return chainNodes
+}
+
+/**
+ * Determines whether a rel cons can be set between {@code node1} and {@code node2}
+ * @param node1 One of the nodes
+ * @param node2 The other one of the nodes
+ */
+export function forbiddenRC(node1: KNode, node2: KNode): boolean {
```

Comment:
consider refactoring the function name to something more clear

Reply:
Maybe I should do that for all variables that are named something abbreviated.